modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheConfigurationEnrichment.java

```diff
@@ -93,4 +95,23 @@ public boolean hasField(String name) {
     @Override public String toString() {
         return S.toString(CacheConfigurationEnrichment.class, this);
     }
+
+    /** {@inheritDoc} */
+    @Override public boolean equals(Object o) {
```

Comment:
I think we can't resue `ClusterCachesInfo#validateCacheGroupConfiguration` here.
We must be sure that two object absolutely equals (because they can't be changed between two snapshots).

And AFAICU `ClusterCachesInfo#validateCacheGroupConfiguration` method checks that cache configuration are **compatible** to be in specific group.

Reply:
>  As the latter looks exceed - I suppose that difference in the Enrichment fields shouldn't affect snapshot.

I thought we want prohibit any changes in cache configuration between snapshots.
If so - equal comparsion looks most natural.